setup.py

```diff
@@ -25,6 +27,17 @@
 if py_version >= (3, 5):
     packages.append('websockets/py35')
 
+ext_modules = []
+if (platform.python_implementation() == 'CPython' and
+        os.environ.get('WEBSOCKETS_CEXT_BUILD') != '0'):
```

Comment:
I settled for writing the Python version and the C version of this function in separate module, testing them both identically, and just try/catch the import of the C version.

I don't mind too much if the try/catch isn't covered. I excluded it from coverage.